### PR TITLE
created nvidia gpu operator addon

### PR DESCRIPTION
**Pull request by Howlla**
*Issue #, if available:*

*Description of changes:* 
Creates new addon for gpu-operator which extends helm addon. Still working on md file for documentation.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: lib/addons/gpu-operator/index.ts
Comment (elamaran11) [inline lib/addons/gpu-operator/index.ts:31]: Please add some jsdoc on what this addon is doing?

File: lib/addons/gpu-operator/index.ts
Comment (elamaran11) [inline lib/addons/gpu-operator/index.ts:23]: Is this the latest version?

**Review by elamaran11 — CHANGES_REQUESTED** [review]
@Howlla Thankyou for taking time to get this addon. Couple of things are missing
1. Addon Documentation is missing. Check `docs/addons/grafana-operator.md` as example.
2. Addon is missing in indexing list of documentation.
3. Addon should be added to e2e in `examples/blueprint-construct/index.ts`
4. `mkdocs.yaml` should be updated with new addon.
5. Also see if you have opportunity to add unit tests under `tests`

overall great work. Are you planning to do `gpu-builder` in separate PR?

File: lib/addons/gpu-operator/index.ts
Comment (Howlla) [reply lib/addons/gpu-operator/index.ts]: updated

**Comment by Howlla** [comment]
Created builder and addon with documentation. Also added it to the e2e test 

@elamaran11 i was not sure whether to add the gpu operator addon in the builder itself or will we do it through the addon. It is a small change in line 124 of `gpu-builder.ts`

File: lib/builders/gpu-builder.ts
Comment (elamaran11) [inline lib/builders/gpu-builder.ts:120]: I think you should have GPU Addon as part of the builder. You should also be adding other required addons for the cluster which you used in your sample runs like coredns, kube-proxy etc etc to minimally run the solution. You can follow observability builder as an example to have a method like `enableGPU` and have all addons. Also as i mentioned before, you should think about grabbing all values as parameters from pattern you are building to builder class to pass to GPU Addon like AmpAddon.

File: mkdocs.yml
Comment (elamaran11) [inline mkdocs.yml:49]: Builder class should also be updated to doc.

**Review by elamaran11 — CHANGES_REQUESTED** [review]
@Howlla Looks good to me, great work on the Builder class on Addon. Following needs to be done :
1. Builder class doc is missing
2. Also builder class should be added to mkdocs. 
3. Plus the builder class is not aligned to functionality we discussed yesterday. So please make sure you add GPU Addon, have a method like `enableGPU` and parametrize GPU like AmpAddon.

File: lib/builders/gpu-builder.ts
Comment (elamaran11) [inline lib/builders/gpu-builder.ts:119]: Why is this needed? We added this specifically for windows builder to work. If you check the Observability builder, we dont have this.

**Review by elamaran11 — CHANGES_REQUESTED** [review]
@Howlla We are on track. Just couple of comments.

File: lib/builders/gpu-builder.ts
Comment (Howlla) [reply lib/builders/gpu-builder.ts]: My bad, i was taking argument into builder rather than into the enablegpu function. fixed this

File: docs/builders/gpu-builder.md
Comment (Howlla) [reply docs/builders/gpu-builder.md]: Fixed, should i remove the nodeRole arguments from gpu-builder.ts?

Comment (elamaran11) [reply docs/builders/gpu-builder.md]: `nodeRole` itself is not required here because its taken care by blueprints. This was specific to make windows work. I think you should follow the Observability Builder convention.

**Review by elamaran11 — COMMENTED** [review]
@shapirov103 This PR looks near to complete with couple of comments open. Can you take a look in the PR, provide feedback so we can merge this with Release?

File: .github/workflows/linkcheck.json
Comment (shapirov103) [inline .github/workflows/linkcheck.json:26]: minor: line 26 is duplicate of line 24

File: lib/builders/gpu-builder.ts
Comment (shapirov103) [inline lib/builders/gpu-builder.ts:65]: there is a warning the defaultOptions are assigned value but never used. 
1. make sure `make lint` is success
2. Why do we have all the default options specified but never use them? is it an indication of a defect?

File: lib/builders/gpu-builder.ts
Comment (shapirov103) [inline lib/builders/gpu-builder.ts:119]: let's get the ID, I will merge it then.

File: lib/builders/gpu-builder.ts
Comment (shapirov103) [inline lib/builders/gpu-builder.ts:89]: what are these values? it is going confusing to the customers which values are allowed to be passed. the way it stands, it is just key value pairs and no indication of the schema. Did you mean to allow passing some particular type here?

File: lib/builders/gpu-builder.ts
Comment (shapirov103) [inline lib/builders/gpu-builder.ts:103]: Any constructor args to allow customers to pass here?

File: docs/addons/gpu-operator.md
Comment (shapirov103) [inline docs/addons/gpu-operator.md:26]: Let's create an explicit type for these values similar to what we did for example for container insights [here](https://github.com/aws-quickstart/cdk-eks-blueprints/blob/main/lib/addons/container-insights/values.ts).

**Review by shapirov103 — COMMENTED** [review]
@Howlla great work, I added a few comments, there is a bit of work left, happy to discuss. 
Please also make sure you run standard commands for PR submission such as `make lint`. This process is described here: https://github.com/aws-quickstart/cdk-eks-blueprints/blob/main/docs/internal/readme-internal.md#submitting-pull-requests

File: lib/builders/gpu-builder.ts
Comment (Howlla) [reply lib/builders/gpu-builder.ts:68]: Fixed

File: lib/builders/gpu-builder.ts
Comment (Howlla) [reply lib/builders/gpu-builder.ts]: Added valueSchema similar to container insights values

File: lib/builders/gpu-builder.ts
Comment (Howlla) [reply lib/builders/gpu-builder.ts]: Passing gpuOptions now

File: docs/addons/gpu-operator.md
Comment (Howlla) [reply docs/addons/gpu-operator.md:26]: Created

File: lib/builders/gpu-builder.ts
Comment (elamaran11) [reply lib/builders/gpu-builder.ts:133]: @Howlla Use this `qs-1ug68anj6` id

Comment (Howlla) [reply lib/builders/gpu-builder.ts]: done

**Comment by Howlla** [comment]
Fixed the issues @shapirov103

File: lib/builders/gpu-builder.ts
Comment (elamaran11) [inline lib/builders/gpu-builder.ts:131]: this should not be exported.

File: lib/builders/gpu-builder.ts
Comment (elamaran11) [inline lib/builders/gpu-builder.ts:1]: remove the global blueprints reference and use individual classes. Check `observability-builder.ts` for reference in current PR.

**Review by elamaran11 — CHANGES_REQUESTED** [review]
@Howlla The `gpu-builder.ts` should be reworked based on my current PR for observability, Also you should add the builder to `index.ts` in `builder` folder in lib. Please let us know once done, we can review.

**Comment by Howlla** [comment]
made changes @elamaran11

**Comment by Howlla** [comment]
I havent merged from your yours but i made sure no conflicts occur. Let me know if you want me to create a branch with both ghaction and gpu builder code

**Comment by elamaran11** [comment]
> I havent merged from your yours but i made sure no conflicts occur. Let me know if you want me to create a branch with both ghaction and gpu builder code

You can do couple of things. One merge from my branch or merge from main after my PR is merged.

**Comment by Howlla** [comment]
Merged with main branch and resolved conflicts

**Review by elamaran11 — APPROVED** [review]
LGTM. @shapirov103 Please check and merge this.,

File: lib/builders/gpu-builder.ts
Comment (elamaran11) [inline lib/builders/gpu-builder.ts:1]: this should be `../stacks`. Same applies to addons, utils, spi and clusterproviders

**Review by elamaran11 — CHANGES_REQUESTED** [review]
Some more minor feedback.

**Review by elamaran11 — DISMISSED** [review]
LGTM. @shapirov103 Please review this, we can move to merge.

File: examples/blueprint-construct/index.ts
Comment (shapirov103) [inline examples/blueprint-construct/index.ts:193]: is this a mandatory parameter? I am curious how customers (and us) will update this moving forward.

**Review by shapirov103 — DISMISSED** [review]
@Howlla Great work! 
I added a question that should not affect the PR status, i will merge after e2e.

File: examples/blueprint-construct/index.ts
Comment (Howlla) [reply examples/blueprint-construct/index.ts:193]: This toolkit version was verified as working with AL2 by Re Alvarez

**Pull request merged**
